packages/stark-ui/src/modules/table/components/table.component.ts

```diff
@@ -0,0 +1,595 @@
+			itemStr = JSON.stringify(item);
+		} else {
+			itemStr = item;
+		}
+
+		return filterCriteria.filter((criteria: RegExp) => {
+			return !criteria.test(itemStr); // the item does not fulfill the given criteria
+		});
+	}
+
+	/**
+	 * Normalize the given filter value into a more complex filter criteria supporting wildcards
+	 * @param filterValue - The original filter value
+	 * @returns An array containing more complex regex based on the original filter value supporting wildcards
+	 */
+	public getNormalizedFilterCriteria(filterValue: string): RegExp[] {
+		const filter: string = filterValue
+			.replace(/\\(?=\*)\*/g, "<stark_char_star>") // string "\*" (escaped *)
+			.replace(/\\(?=\?)\?/g, "<stark_char_quot>") // string "\?" (escaped ?)
+			.replace(/\\/g, "<stark_char_backsl>") // character "\"
+			.replace(/[*?\[\]()$+^]/g, (match: string) => {
+				// replace chars "*", "?", "[", "]", "(", ")", "$", "+", "^"
+				if (match === "*") {
+					return "\\S*"; // wildcard "*"
+				} else if (match === "?") {
+					return "\\S{1}"; // wildcard "?"
+				} else {
+					return "\\" + match; // add trailing "\" to escape the character
+				}
+			})
+			.replace("<stark_char_star>", "\\*")
+			.replace("<stark_char_quot>", "\\?")
+			.replace("<stark_char_backsl>", "\\\\");
+
+		return filter.split(" ").map((filterStr: string) => new RegExp(filterStr, "i"));
+	}
+
+	/**
+	 * Return whether the number of selected elements matches the total number of rows.
+	 */
+	public isAllSelected(): boolean {
+		const numSelected: number = this.selection.selected.length;
+		const numRows: number = this.data.length;
+		return numSelected === numRows;
+	}
+
+	/**
+	 * Called when the Clear button in the filter pop-up is clicked
+	 */
+	public onClearFilter(): void {
+		this.filter.globalFilterValue = "";
+		this.applyFilter();
+	}
+
+	/**
+	 * Called whenever the value of the filter input changes
+	 */
+	public onFilterChange(): void {
+		this.applyFilter();
+	}
+
+	/**
+	 * Called whenever the sorting of any of the columns changes
+	 * @param column - The column whose sorting has changed
+	 */
+	public onReorderChange(column: StarkTableColumnComponent): void {
+		if (column.sortable) {
+			this.resetSorting(column);
+			if (column.sortable) {
+				column.sortPriority = 1;
+				switch (column.sortDirection) {
+					case "asc":
+						column.sortDirection = "desc";
+						break;
+					case "desc":
+						column.sortDirection = "";
+						break;
+					default:
+						column.sortDirection = "asc";
+						break;
+				}
+			}
+			this.sortData();
+		}
+	}
+
+	/**
+	 * Open the multi-sort dialog to configure the multi-column sorting of the data
+	 */
+	public openMultiSortDialog(): void {
+		const dialogRef: MatDialogRef<StarkTableMultisortDialogComponent, StarkSortingRule[]> = this.dialogService.open<
+			StarkTableMultisortDialogComponent,
+			StarkTableMultisortDialogData
+		>(StarkTableMultisortDialogComponent, {
+			width: "500px",
```

Comment:
We should try to set this width through the SCSS, shall we create a ticket for that?

Reply:
Ok. A ticket is not necessary, I'll fix it 😉